packages/hooks/src/createHasHook.js

```diff
@@ -5,17 +5,25 @@
  * @param  {Object}   hooks Stored hooks, keyed by hook name.
  *
  * @return {Function}       Function that returns whether any handlers are
- *                          attached to a particular hook.
+ *                          attached to a particular hook and optional namespace.
  */
 function createHasHook( hooks ) {
 	/**
-	 * Returns how many handlers are attached for the given hook.
+	 * Returns whether any handlers are attached for the given hookName and optional namespace.
 	 *
-	 * @param  {string}  hookName The name of the hook to check for.
+	 * @param {string} hookName  The name of the hook to check for.
+	 * @param {string} namespace Optional. The unique namespace identifying the callback in the form `vendor/plugin/function`.
 	 *
 	 * @return {boolean} Whether there are handlers that are attached to the given hook.
 	 */
-	return function hasHook( hookName ) {
+	return function hasHook( hookName, namespace ) {
+		// Use the namespace if provided.
+		if ( 'undefined' !== typeof namespace ){
+			return hookName in hooks &&
+				hooks[ hookName ].handlers &&
+				hooks[ hookName ].handlers.map( hook => hook['namespace'] ).includes( namespace );
```

Comment:
`Array#includes` is not polyfilled, and is not supported in older browsers:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Browser_compatibility

Further, there's unnecessary work here to iterate the entire array of handlers, when `Array#some` serves this purpose to terminate iteration once the first match is encountered:

```js
hooks[ hookName ].handlers.some( hook => hook['namespace'] === namespace );
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
Ah, right - thanks for catching that. Will fix.

Reply:
updated